File: implement_strstr.py
class Solution:
    # @param haystack, a string
    # @param needle, a string
    # @return an integer
    def strStr(self, haystack, needle):
        n=len(haystack)
        m=len(needle)
        if n<m:
            return -1
        print n,m
        for x in range (n-m+1):
            print haystack[x:m+x],needle
            if haystack[x:m+x]==needle:
                return x                
        return -1
    def 
            

if __name__=='__main__':
    sol=Solution()
    haystack='mississippi'
    needle='pi'
    print sol.strStr(haystack,needle)

File: Algorithm/CPP/implement_strstr.cpp
# include <iostream>
# include <string>
using namespace std;



class Solution {
public:
	int strStr(string haystack, string needle) {
		if (needle.size() > haystack.size())
			return -1;
		for (auto i = 0; i <= (haystack.size()-needle.size()); ++i)
		{
			if (haystack.substr(i, needle.size()) == needle)
				return i;
		}
		return -1;


	}
};


int main()
{
	string haystack = "abced";
	string needle = "bc";
	Solution sol;
	cout << sol.strStr(haystack, needle);
	cin.get();
	return 0;
}
